libbeat/autodiscover/providers/kubernetes/kubernetes.go

```diff
@@ -154,6 +155,44 @@ func (p *Provider) publish(event bus.Event) {
 	// Try to match a config
 	if config := p.templates.GetConfig(event); config != nil {
 		event["config"] = config
+	} else {
```

Comment:
this code path should be similar `p.templates.GetConfig`, I would extract all the logic to generate hints from the event, and then call `p.builders.GetConfig`

Reply:
it may be a bit difficult as handling is provider specific unlike templates. WDYT?

Reply:
done